tracing-subscriber/src/registry/mod.rs

```diff
@@ -45,3 +54,96 @@ pub trait LookupMetadata {
         self.metadata(id).is_some()
     }
 }
+
+pub trait LookupSpan<'a> {
+    type Data: SpanData<'a>;
+    fn span_data(&'a self, id: &Id) -> Option<Self::Data>;
+
+    fn span(&'a self, id: &Id) -> Option<SpanRef<'_, Self>>
```

Comment:
is there a reason this returns a `SpanRef<'_, Self>` instead of a `SpanRef<'a, Self>`? Seems kinda surprising to me...